cibuildwheel/__main__.py

```diff
@@ -341,11 +346,7 @@ def print_preamble(platform: str, options: Options, identifiers: list[str]) -> N
 def get_build_identifiers(
     platform: PlatformName, build_selector: BuildSelector, architectures: set[Architecture]
 ) -> list[str]:
-    python_configurations: (
-        list[cibuildwheel.linux.PythonConfiguration]
-        | list[cibuildwheel.windows.PythonConfiguration]
-        | list[cibuildwheel.macos.PythonConfiguration]
-    )
+    python_configurations: typing.Sequence[GenericPythonConfiguration]
```

Comment:
```suggestion
    python_configurations: Sequence[GenericPythonConfiguration]
```

Reply:
Ahh, can't add `from collections.abc import Sequence` as a suggestion. This is fine for now, but ideally we should do that (also `Set`, from there, for input arguments, is better than a concrete and mutable `set`).

Reply:
oh yeah it hadn't occurred to me that `set[A]` is similar to `MutableSet[A]`? but still more specific? But `Dict` is the same as `dict` and `Map` / `MutableMap` are different?

Reply:
`Dict` is just a older way to say `dict[...]`. I think `typing.Set` / `set` are also similar, while `collections.abc.{Map,MutableMap,Set,MutableSet}` are more general structural types. Input parameters should generally be as loose as possible, and output parameters should be as specific as possible (usually concrete).